solutions/ex3/Main.hs

```diff
@@ -202,6 +216,22 @@ createSortInclusionInfer ''VarL    ''LhsL
 createSortInclusionInfer ''ExpL    ''RhsL
 createSortInclusionInfer ''AssignL ''StatementL
 
+--- Question: why is it enough to just convert AssignL to StatementL as sorts?
+-- An `Assign` constructor has more data than a `Statement` constructor.
```

Comment:
When you add an injection from AssignL into StatementL, then Assign becomes a constructor that can be ascribed the StatementL sort. It is thus tautologically false to say that an Assign constructor has more data than a Statement constructor, assuming that you're referring to the StatementL sort and not the Statement fragment.

Reply:
Further, since the only AssignOp available in Imp3 is AssignOpEquals, it's false to say that  the generic Assign constructor has more information than the Imp3-specific Assign constructor.